internal/controller/service_controller.go

```diff
@@ -294,6 +297,10 @@ func (r *ServiceReconciler) applyPolicy(nbResource *netbirdiov1.NBResource, svc
 	return nil
 }
 
+func (r *ServiceReconciler) routingPeerName() string {
+	return fmt.Sprintf("%s-%s", r.RouterNamePrefix, routingPeerBaseName)
```

Comment:
This is causing the router name to be `-router` when prefix is empty
```suggestion
	if r.RouterNamePrefix == "" {
		return routingPeerBaseName
	}
	return fmt.Sprintf("%s-%s", r.RouterNamePrefix, routingPeerBaseName)
```